lib/HttpClient.ts

```diff
@@ -91,7 +91,8 @@ export class HttpClient implements ifm.IHttpClient {
     private _httpProxy: ifm.IProxyConfiguration;
     private _httpProxyBypassHosts: RegExp[];
     private _allowRedirects: boolean = true;
-    private _maxRedirects: number = 50
+    private _allowRedirectDowngrade: boolean = false;
```

Comment:
I want to call this out as a breaking change. This is fine IMO, since we've decided this is a security vulnerability (which I agree with to be clear), and are now considering the downgrading behavior to be a bug.

Reply:
I think it's fine since it's calling out an odd and less secure behavior.